flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/enumerator/DynamoDbStreamsSourceEnumerator.java

```diff
@@ -104,6 +104,11 @@ public DynamoDbStreamsSourceEnumerator(
     public void start() {
         context.callAsync(this::discoverSplits, this::processDiscoveredSplits);
         final long shardDiscoveryInterval = sourceConfig.get(SHARD_DISCOVERY_INTERVAL).toMillis();
+        context.callAsync(
+                this::incrementallyDiscoverSplits,
+                this::processDiscoveredSplits,
+                0,
+                shardDiscoveryInterval);
```

Comment:
should we separate out the configuration for incremental shard discovery interval and the full shard discovery interval?

Reply:
yes, makes sense, i updated it to a new config